hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestWriteCopyOnWrite.java

```diff
@@ -169,8 +169,10 @@ public void testPartialFailover() throws Exception {
         .checkpointThrows(4, "Timeout(1000ms) while waiting for instant initialize")
         .assertEmptyEvent()
         .subTaskFails(0, 3)
-        // the last checkpoint instant was rolled back by subTaskFails(0, 2)
-        // with EAGER cleaning strategy
+        // the last checkpoint instant can not rolled back by subTaskFails(0, 2)
```

Comment:
`can not rolled` -> `can not be rolled`

Reply:
And while we introduce new changes, also ensure the original test case is covered instead of just modifying the test to pass.

Reply:
Agree. I added a new test and gave some reasons for the changes to the original test:
In `INSERT` mode, only job-level restart can roll back the pending Instant. When SubTasks fail to restart, the pending Instant needs to be reused.